ext/standard/string.c

```diff
@@ -4006,7 +4007,9 @@ static zend_long php_str_replace_in_subject(zval *search, zval *replace, zval *s
 				/* Get current entry */
 				if ((replace_entry = zend_hash_get_current_data_ex(Z_ARRVAL_P(replace), &pos)) != NULL) {
 					/* Make sure we're dealing with strings. */
-					convert_to_string_ex(replace_entry);
+					ZVAL_DEREF(replace_entry);  // Deal with possibly shared zvals
```

Comment:
Just noticed that `zval_get_string` won't cause the same behavior from PHP-5.6, and backwards, since `string_replace` wouldn't cast each array val (from $search and $replaces args) to string anymore.

Should we change that collateral behavior?